maven-core/src/main/java/org/apache/maven/execution/MavenExecutionRequest.java

```diff
@@ -494,14 +504,51 @@ public interface MavenExecutionRequest {
 
     /**
      * @since 3.3.0
+     * @deprecated use {@link #setRootdir(Path)} instead
      */
+    @Deprecated
     void setMultiModuleProjectDirectory(File file);
 
     /**
      * @since 3.3.0
+     * @deprecated use {@link #getRootdir()} instead
      */
+    @Deprecated
     File getMultiModuleProjectDirectory();
 
+    /**
+     * Sets the top dir of the project.
+     *
+     * @since 4.0.0
+     */
+    MavenExecutionRequest setTopdir(Path topdir);
```

Comment:
dir --> Directory, here and elsewhere

That is, don't abbreviate.

Reply:
That is everywhere the case and the properties end with `dir` as well.

Reply:
It's not the case in the existing methods like setBaseDirectory and it's not the Java approach. Non-abbreviated names are more readable and clear.

Reply:
Then we need to review everything in Core and make it consistent.

Reply:
That would be nice, but it's not a prerequisitie for the choosing the more readable name here.